### PR TITLE
[onert/train] Register LayerScopeTensor to registry

### PR DESCRIPTION
**Pull request by zetwhite**
This PR registers LayerScopeTensor from each layer into tensor registry.

ONE-DCO-1.0-Signed-off-by: seunghui youn <sseung.youn@samsung.com>

--------------------------------------

draft : https://github.com/Samsung/ONE/pull/13486

### CODE REVIEW

File: runtime/onert/backend/train/BackendContext.cc [hunk 1]
```diff
@@ -262,7 +262,43 @@ FunctionMap BackendContext::generateFunctionMap()
...
+    if (not trainable_op->isRequiredForBackward())
+      continue;
+
+    VERBOSE(BackendContext) << "register layerscope tensor for " << trainable_op->name()
+                            << std::endl;
+
+    fn_seq->iterate([&](exec::train::ITrainableFunction &fn) {
+      register_tensors(op_idx, (&fn)->registerLayerScopeTensors());
+    });
+  }
```
Comment (zetwhite) [inline runtime/onert/backend/train/BackendContext.cc:301]: I'll show log like this : 

```bash 
[ BackendContext ] register layerscope tensor for CategoricalCrossentropyLoss
[ BackendContext ] register layerscope tensor for Softmax
[ BackendContext ] register layerscope tensor for FullyConnected
[ BackendContext ] (idx:  @8-0) registered
[ BackendContext ] (idx:  @8-1) registered
[ BackendContext ] (idx:  @8-2) registered
[ BackendContext ] register layerscope tensor for FullyConnected
[ BackendContext ] (idx:  @7-0) registered
[ BackendContext ] (idx:  @7-1) registered
[ BackendContext ] (idx:  @7-2) registered
[ BackendContext ] (idx:  @7-3) registered
[ BackendContext ] register layerscope tensor for Reshape
[ BackendContext ] register layerscope tensor for MaxPool2D
[ BackendContext ] (idx:  @5-0) registered
[ BackendContext ] register layerscope tensor for Conv2D
[ BackendContext ] (idx:  @4-0) registered
[ BackendContext ] (idx:  @4-1) registered
[ BackendContext ] (idx:  @4-2) registered
[ BackendContext ] (idx:  @4-3) registered
[ BackendContext ] register layerscope tensor for Conv2D
[ BackendContext ] (idx:  @3-0) registered
[ BackendContext ] (idx:  @3-1) registered
[ BackendContext ] (idx:  @3-2) registered
[ BackendContext ] (idx:  @3-3) registered
```

File: runtime/onert/backend/train/BackendContext.cc [hunk 2]
```diff
@@ -262,7 +262,43 @@ FunctionMap BackendContext::generateFunctionMap()
...
+    assert(trainable_op != nullptr);
+
+    if (not trainable_op->isRequiredForBackward())
+      continue;
+
+    VERBOSE(BackendContext) << "register layerscope tensor for " << trainable_op->name()
+                            << std::endl;
+
+    fn_seq->iterate([&](exec::train::ITrainableFunction &fn) {
+      register_tensors(op_idx, (&fn)->registerLayerScopeTensors());
```
Comment (zetwhite) [inline runtime/onert/backend/train/BackendContext.cc:299]: `(&fn)->registerLayerScopeTensors()` call this functions and each layer will override this function. 

https://github.com/Samsung/ONE/blob/190b538fb220479ef22c8a8180f93be24a83ca73/runtime/onert/core/include/exec/train/ITrainableFunction.h#L39-L43

**Review by ragmani — APPROVED** [review]
LGTM

**Comment by zetwhite** [comment]
@glistening Could you merge this PR?

**Pull request merged**